src/main/java/org/jitsi/videobridge/cc/vp8/VP8AdaptiveTrackProjectionContext.java

```diff
@@ -255,17 +311,26 @@ VP8FrameProjection createVP8FrameProjection(
             return null;
         }
 
+        if (lastVP8FrameProjection.getSSRC() != nextVP8FrameProjection.getSSRC())
+        {
+            /* We've changed the encoding we're forwarding, so previous frames
+             * in the map are no longer relevant.
+             */
+            vp8FrameProjectionMap.clear();
```

Comment:
i can't remember, is the logic which handles the layer switching elsewhere? that is, we don't have to worry about 'finishing' a frame of a previous layer here, and therefore keeping its projection around, when we see a new ssrc?

Reply:
The logic which handles layer switching is in `VP8QualityFilter`, which is why I have to do things in this somewhat indirect way.

We can only switch layers on a keyframe, so it's safe to abandon frames from before the switch. In some cases (e.g. dropping to a lower bandwidth layer due to bandwidth constraints) it's even helpful to do so.

Reply:
ok, i thought there was some scenario which was like "we want to switch to this, but need to get the full keyframe" or something and there was a transition where we needed to keep both alive at once, but my memory is fuzzy so don't know for sure.

Reply:
As far as I can tell, frames are only put in the projection map once they're actually ready to be routed, which implies we're actually switching.  But I'll double-check the flow.